### PR TITLE
MAKE-680: clone CreateOptions before spawning new processes with them

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/MAKE-680

## Changes
* Added copy function.
* Copied all CreateOptions in any place that spawns new processes.
* Added more documentation for CreateOptions.

### CODE REVIEW

File: create.go [hunk 1]
```diff
@@ -191,3 +191,48 @@ func (opts *CreateOptions) Close() error {
 	}
 	return catcher.Resolve()
 }
+
+// Copy returns a copy of the exported fields in CreateOptions. The caller is
+// responsible for re-populating the unexported fields, which are cleared.
+func (opts *CreateOptions) CopyExported() *CreateOptions {
```
Comment (tychoish) [inline create.go]: I'd call this Copy or Export

File: output.go [hunk 2]
```diff
@@ -411,3 +411,21 @@ func (o *OutputOptions) GetError() (io.Writer, error) {
 
 	return o.errorMulti, nil
 }
+
+// CopyExported returns a copy of the options for only the exported fields.
+// Unexported fields are cleared.
+func (o *OutputOptions) CopyExported() *OutputOptions {
```
Comment (tychoish) [inline output.go]: copy or export

File: create.go (hunk 1 above)
Comment (Kimchelly) [reply create.go]: I renamed this to Copy.

File: output.go (hunk 2 above)
Comment (Kimchelly) [reply output.go]: Copy

**Pull request merged**